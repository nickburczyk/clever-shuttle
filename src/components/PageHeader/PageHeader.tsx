import styles from './styles.module.css'
import { useNavigate } from 'react-router-dom'

interface Props {
    title: string;
}

const PageHeader = ({title}:Props) => {
    const nav = useNavigate()

    return(
        <div className={styles.heading}>
            <button className='btn secondary' onClick={()=>nav(-1)}>{" < "}</button>
            <h1>{title}</h1>
        </div>
    )
}

export default PageHeader