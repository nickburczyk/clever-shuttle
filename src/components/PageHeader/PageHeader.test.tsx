import { create } from 'react-test-renderer'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'

import PageHeader from "./PageHeader";

describe("Search Box", () => {
    const history = createMemoryHistory({ initialEntries: ['/'] });

    it ("should render with title `Hello World`", () => {
        const tree = create(
            <Router navigator={history} location={history.location}>
                <PageHeader title={"Hello World!"}/>
            </Router>
        )
        expect(tree.toJSON()).toMatchSnapshot()
    })
    
    it ("should render with title `Welcome to the thunderdome!`", () => {
        const tree = create(
            <Router navigator={history} location={history.location}>
                <PageHeader title={"Welcome to the thunderdome"}/>
            </Router>
        )
        expect(tree.toJSON()).toMatchSnapshot()
    })
})