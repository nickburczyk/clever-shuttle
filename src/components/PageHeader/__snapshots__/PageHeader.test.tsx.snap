// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box should render with title \`Hello World\` 1`] = `
<div
  className="heading"
>
  <button
    className="btn secondary"
    onClick={[Function]}
  >
     &lt; 
  </button>
  <h1>
    Hello World!
  </h1>
</div>
`;

exports[`Search Box should render with title \`Welcome to the thunderdome!\` 1`] = `
<div
  className="heading"
>
  <button
    className="btn secondary"
    onClick={[Function]}
  >
     &lt; 
  </button>
  <h1>
    Welcome to the thunderdome
  </h1>
</div>
`;
